src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -451,6 +451,26 @@ private function _process_directives( string $html ) {
 
 					// Checks if there is a server directive processor registered for each directive.
 					foreach ( $p->get_attribute_names_with_prefix( 'data-wp-' ) as $attribute_name ) {
+						if ( ! preg_match(
+							/*
+							 * This must align with the client-side regex used by the interactivity API.
+							 * @see https://github.com/WordPress/gutenberg/blob/ca616014255efbb61f34c10917d52a2d86c1c660/packages/interactivity/src/vdom.ts#L20-L32
```

Comment:
This is really helpful, but can probably be improved since this URL is pinned to a specific hash. Not a blocker for commit though.